### PR TITLE
19/WAKU2-LIGHTPUSH: Update

### PR DESCRIPTION
**Pull request by jimstir**
Updating the 19/WAKU2-LIGHTPUSH rfc. Waku specs repo has a new rfc becuase the current specification is under specified. Information from that update is included with some changes. Also introduced the term edge nodes in this update.

### CODE REVIEW

File: waku/standards/core/19/lightpush.md [hunk 1]
```diff
@@ -11,65 +11,115 @@ contributors:
...
-If they are unable to do so for some reason,
-they SHOULD return an error code in `PushResponse`.
+Nodes that respond to `LightPushRequest` SHOULD
+
+- either relay the encapsulated message via
+[11/WAKU2-RELAY](/waku/standards/core/11/relay.md) protocol on the specified `pubsub_topic`
+- or perform another requested service.
+`Services beyond [11/WAKU2-RELAY](/waku/standards/core/11/relay.md) are yet to be defined.`
+
+Depending on the network configuration, the lightpush client may not need to provide `pubsub_topic`.
```
[the remark this answers is not in the document]
Comment (jm-clius) [reply waku/standards/core/19/lightpush.md:69]: Within the context of `lightpush` and other req-resp protocols the agents are always _client_ and _service node_ ("server" is sometimes used though somewhat less accurate IMO). I would stay away from "light node"/"edge node" within the context of req-resp protocol specifications. In other words, "lightpush client" would be correct in this context.

**Review by jm-clius — COMMENTED** [review]
Without having reviewed in detail, I'm wondering if this update is premature? Waku specs repo indeed has a new version of the lightpush spec as experimental improvement, but I would suggest that the implementation and deployment reach some level of maturity before we replace the existing RFC (as this one is already in `draft` and matches the implementation). Both versions of the protocol will be supported for a while @NagyZoltanPeter can comment on maturity of the "new" protocol implementation. Once most clients have switched over, we can deprecate the old lightpush version - both the RFC and the implementation.

**Comment by NagyZoltanPeter** [comment]
> Without having reviewed in detail, I'm wondering if this update is premature? Waku specs repo indeed has a new version of the lightpush spec as experimental improvement, but I would suggest that the implementation and deployment reach some level of maturity before we replace the existing RFC (as this one is already in `draft` and matches the implementation). Both versions of the protocol will be supported for a while @NagyZoltanPeter can comment on maturity of the "new" protocol implementation. Once most clients have switched over, we can deprecate the old lightpush version - both the RFC and the implementation.

Thanks @jm-clius, yes. new redesigned v2 of lightpush protocol is under development, spec available here:  https://github.com/waku-org/specs/blob/master/standards/core/lightpush.md
cc: @jimstir, of course any comments are welcome.

**Comment by NagyZoltanPeter** [comment]
> > Without having reviewed in detail, I'm wondering if this update is premature? Waku specs repo indeed has a new version of the lightpush spec as experimental improvement, but I would suggest that the implementation and deployment reach some level of maturity before we replace the existing RFC (as this one is already in `draft` and matches the implementation). Both versions of the protocol will be supported for a while @NagyZoltanPeter can comment on maturity of the "new" protocol implementation. Once most clients have switched over, we can deprecate the old lightpush version - both the RFC and the implementation.
> 
> Thanks @jm-clius, yes. new redesigned v2 of lightpush protocol is under development, spec available here: https://github.com/waku-org/specs/blob/master/standards/core/lightpush.md cc: @jimstir, of course any comments are welcome.

@jimstir sorry I did not see your changes are based upon that spec... but yes, we are planning to have a slow rollover to the new protocol, leaving enough time to switch.

**Comment by jimstir** [comment]
Thanks for the feedback. Closing this pr, as this RFC will become deprecated in the future and replaced with the new light push protocol located https://github.com/waku-org/specs/blob/master/standards/core/lightpush.md.

**Pull request closed without merging**